src/mcp/client/stdio/__init__.py

```diff
@@ -173,10 +172,9 @@ async def stdin_writer():
             yield read_stream, write_stream
         finally:
             # Clean up process to prevent any dangling orphaned processes
-            if sys.platform == "win32":
-                await terminate_windows_process(process)
-            else:
-                process.terminate()
+            tg.cancel_scope.cancel()
+            process.terminate()
```

Comment:
This won't work if the process doesn't react to SIGTERM. We need a timeout and a fallback on SIGKILL.
Please test it with https://github.com/modelcontextprotocol/servers/blob/main/src/everything/index.ts#L13 (windows & unix)

Reply:
Ref: https://github.com/modelcontextprotocol/python-sdk/pull/555